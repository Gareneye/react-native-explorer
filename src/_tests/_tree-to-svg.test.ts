// import * as fs from "fs"
import * as yoga from "yoga-layout"

import treeToSVG from "../tree-to-svg"

describe("treeToSVG", () => {
    it("wraps whatever text you pass into it with an SVG schema", () => {
      const rootNode = yoga.Node.create()
      rootNode.setWidth(600)
      rootNode.setHeight(400)
      rootNode.setMargin(yoga.EDGE_ALL, 20)
      rootNode.setDisplay(yoga.DISPLAY_FLEX)
      rootNode.setFlexDirection(yoga.FLEX_DIRECTION_ROW)

      const settings = {
        width: 1024,
        height: 768
      }

      const results = treeToSVG(rootNode, settings)
      expect(results).toMatchSnapshot()

      rootNode.freeRecursive()
    })
})
