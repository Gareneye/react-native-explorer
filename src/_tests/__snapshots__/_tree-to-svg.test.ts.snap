// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`treeToSVG wraps whatever text you pass into it with an SVG schema 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<svg width=\\"1024\\" height=\\"768\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\">
  <rect x=\\"20\\" y=\\"20\\" width=\\"600\\" height=\\"400\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\"/>
</svg>
"
`;
